### ISSUE TITLE
Zips made with fflate dont work in jsdos

### ISSUE THREAD

**Issue by DustinBrett** [DustinBrett/daedalOS#81]
Not sure if its related to fflate or the way I made the zips or what, but I couldnt make a zip from within daedalOS that worked properly in jsdos.

**Comment by 101arrowz** [comment]
Author of `fflate` here, I can debug this if you send a reproduction. But from what I can tell [this code](https://github.com/DustinBrett/daedalOS/blob/aad993666e20441f6a47cceb89450d8468091a52/utils/zipFunctions.ts#L6-L18) seems to be doing something weird, you can just create zips like this:

```js
import { zip, strToU8 } from 'fflate';

zip({ 'yourfile.js': strToU8('hello world') }, console.log);
```

**Comment by DustinBrett** [comment]
> Author of `fflate` here, I can debug this if you send a reproduction. But from what I can tell [this code](https://github.com/DustinBrett/daedalOS/blob/aad993666e20441f6a47cceb89450d8468091a52/utils/zipFunctions.ts#L6-L18) seems to be doing something weird, you can just create zips like this:
> 
> ```js
> import { zip, strToU8 } from 'fflate';
> 
> zip({ 'yourfile.js': strToU8('hello world') }, console.log);
> ```
> 
> EDIT: actually it looks like you were creating the directory entries that didn't exist in older versions of `fflate`, the issue has been fixed in v0.7.3 so you can probably remove that code.

Thanks for the comment! I was actually just looking at this again today and it seems to be an issue in the way I was making the zip object for nested folders. I think you've indeed pinpointing where I need to adjust.

**Comment by DustinBrett** [comment]
I had to change a few things I was doing as I was treating nested files improperly. I've done some adjustments to get it going https://github.com/DustinBrett/daedalOS/commit/6a696b3aea0819945409256f430a632b0a2a1ccf.

Thanks for checking into this issue @101arrowz!

**Comment by DustinBrett** [comment]
> Author of `fflate` here, I can debug this if you send a reproduction. But from what I can tell [this code](https://github.com/DustinBrett/daedalOS/blob/aad993666e20441f6a47cceb89450d8468091a52/utils/zipFunctions.ts#L6-L18) seems to be doing something weird, you can just create zips like this:
> 
> ```js
> import { zip, strToU8 } from 'fflate';
> 
> zip({ 'yourfile.js': strToU8('hello world') }, console.log);
> ```
> 
> EDIT: actually it looks like you were creating the directory entries that didn't exist in older versions of `fflate`, the issue has been fixed in v0.7.3 so you can probably remove that code.

Indeed I also was creating those, but I actually forgot why before seeing this EDIT and removed them anyway, haha. Thanks again!
